app/src/main/java/com/example/advanceduiapp/data/MultipleViewRecyclerAdapter.kt

```diff
@@ -0,0 +1,71 @@
+package com.example.advanceduiapp.data
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.*
+import com.example.advanceduiapp.databinding.ButtonItemBinding
+import com.example.advanceduiapp.databinding.FragmentPagerBinding
+import com.example.advanceduiapp.databinding.TvItemBinding
+
+class MultipleViewRecyclerAdapter(
+    private val pageData: List<Pair<String, Int>>,
+    private val textData: List<String>,
+    private val buttonActionsData: List<() -> Unit>
```

Comment:
it's a bad approach. You should have here as much click listeners as required(as a separate properties) and than you put those listeners while you creating viewHolders